packages/core/aqua-compiler/src/index.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    generateSources,
+    generateTypes,
+} from './generate/index.js';
+import { CompilationResult } from './generate/interfaces.js';
+
+type OutputType = 'js' | 'ts';
+
+interface JsOutput {
+    sources: string;
+    types: string;
+}
+
+interface TsOutput {
+    sources: string;
+}
+
+type LanguageOutput = JsOutput | TsOutput;
```

Comment:
Another thought we can use overloaded function to have a predictable output, but instead of that (becuase I know how some folexes hate overloaded functions in typescript) - I would suggest exporting to functions `aquaToJs` and `aquaToTs` - which will have their correspondind return types instead of a union

Returning union is worse because we know EXACTLY what return type is at compile time

Reply:
Made it in another ts hacky way.

Reply:
using genreics is fine as well